metrics/aggregate_test.go

```diff
@@ -277,3 +280,47 @@ func BenchmarkConcurrentAggregate(b *testing.B) {
 		})
 	}
 }
+
+func TestRemoveGaugeMetrics(t *testing.T) {
+	a := NewAggregate()
+
+	// Add some metrics to the aggregate
+	a.parseAndMerge(strings.NewReader(`# TYPE gauge_metric gauge
+gauge_metric{label="value"} 123
+# TYPE counter_metric counter
+counter_metric{label="value"} 456
+`), nil)
+
+	// Simulate a scrape with Prometheus user-agent
+	req, err := http.NewRequest("GET", "/metrics", nil)
+	if err != nil {
+		t.Fatalf("unexpected error creating request: %s", err)
+	}
+	req.Header.Set("User-Agent", "Prometheus/2.0")
+
+	// Create a response recorder
+	rr := httptest.NewRecorder()
+
+	// Handle the request
+	router := gin.Default()
+	router.GET("/metrics", a.HandleRender)
+	router.ServeHTTP(rr, req)
+
+	// Check the response code
+	if status := rr.Code; status != http.StatusOK {
+		t.Errorf("handler returned wrong status code: got %v want %v", status, http.StatusOK)
+	}
+
+	// Check that the gauge metric has been removed
```

Comment:
I like this test, but just for completeness... what do you think about making a second request and check that the gauge is not there?

Not really needed tough, feel free to ignore this comment

Reply:
I don't think it is really needed since the metric no longer exists in the aggregation